### ISSUE TITLE
Duplicate bottom row with fixedRowsBottom

### ISSUE THREAD

**Issue by dosaki** [handsontable/handsontable#4454]
### Description
The fixed bottom row appears twice: once at the end of the rows (as expected) and again at bottom of the container.
It should only appear once (or have an option to disable the one at the bottom of the container)

### Steps to reproduce
1. Create a table with a number of rows that would take up less than the height of the container
2. Set `fixedRowsBottom` to 1
3. Bottom row is duplicated

### Demo
http://jsfiddle.net/ezvLy4cq/

### Your environment
* Handsontable version: Pro 1.13.0
* Browser Name and version: Chromium 60
* Operating System: Linux ArchOS

**Comment by AMBudnik** [comment]
Hi @dosaki 

Thank you. Yes, unfortunately, it is showing like that cause of the CSS settings. The fixed bottom row is n additional 1-row table that duplicates the results of the last row in the main table.

**Comment by dosaki** [comment]
I see. Is there a work-around for this?
All my tables are built dynamically and I don't know how many rows they will have.

Do I have do it programatically, i.e. figure out if all the rows fit in the window and set `fixedRowsBottom` to 0?
If that's the case, I can't find a way to figure out the height of each row reliably from Handson.

**Comment by AMBudnik** [comment]
It would be the easiest way to count rows after table gets rendered (via `countRows` mehod) and then use `height` option inside the `updateSettings` to change the table `height`.

**Comment by dosaki** [comment]
But I still don't know if the last row is visible or not.
I'd need to know how high each row is and I can't see an API to get that.

I suppose I can just get the height of the element though...? What if one of the rows is higher for some reason?

**Comment by AMBudnik** [comment]
@dosaki there's a method called [getLastVisibleRow](https://docs.handsontable.com/pro/1.13.0/AutoRowSize.html#getLastVisibleRow) if it is equal to [instance.countRows()](https://docs.handsontable.com/pro/1.13.0/Core.html#countRows) you'd be able to check if the last row is visible.

**Comment by dosaki** [comment]
I'll give that a go! It sounds like it should work.
Thank you!

**Comment by AMBudnik** [comment]
you're welcome

**Comment by dosaki** [comment]
Here's my work aorund, in case someone stumbles upon this:
```javascript
    var autoRowSizePlugin = handsontableInstance.getPlugin('autoRowSize');
    var totalRows = handsontableInstance.countRows();
    if (autoRowSizePlugin.getLastVisibleRow() === totalRows - 1) {
      handsontableInstance.updateSettings({
        fixedRowsBottom: 0
      });
    }
```

If instead, you want to keep the fixedRowsBottom you can use the bit below:
```javascript
    var autoRowSizePlugin = handsontableInstance.getPlugin('autoRowSize');
    var totalRows = handsontableInstance.countRows();
    if (autoRowSizePlugin.getLastVisibleRow() === totalRows - 1) {
      var totalRowHeight = 0;
      for (var i = 0; i < handsontableInstance.countRows(); i++) {
        totalRowHeight += autoRowSizePlugin.getRowHeight(i, 24);
      }
      document.querySelector(tableContainer).style.height = totalRowHeight + "px";
      handsontableInstance.updateSettings({
        height: totalRowHeight + (autoRowSizePlugin.getColumnHeaderHeight() || 24)
      });
    }
```

**Comment by AMBudnik** [comment]
Thank you for sharing the code @dosaki

**Comment by AMBudnik** [comment]
Replicable in version 0.35.0

The bottom row sticks to the dimensions provided by `height` settings, not the actual table height.
Maybe we could add an exception to move the bottom row up when there are fewer rows than fit into the required table `height`.

![image](https://user-images.githubusercontent.com/10757813/34982001-79c131b0-faa9-11e7-8a6a-f8c85521f73c.png)

**Comment by AMBudnik** [comment]
Update to the issue.

I just created a simple sample app using v`6.2.0`

HTML
```html
<body>
  <div class="box"></div>
  <script src="js.js" charset="utf-8"></script>
</body>
```

JS (`js.js` file)
```js
var box = document.querySelector('.box');

var settings = {
  data: Handsontable.helper.createSpreadsheetData(300, 200),
  colHeaders: true,
  rowHeaders: true,
  fixedRowsBottom: 2
}

var hot = new Handsontable(box, settings);
```

with no additional CSS settings besides the styles provided by `handsontable.full.min.css` and got the same result.

![image](https://user-images.githubusercontent.com/10757813/49222090-39541800-f3db-11e8-9f31-d143d223197d.png)

**Comment by ashish1697** [comment]
I have fixed this issue by CSS. This can help until the fix been made in next release
Hidden both the row header and row data.

#[Your container] .ht_master table tbody tr:last-child, #[Your container] .ht_clone_left table tbody tr:last-child{
display: none;    
}

**Comment by spark889** [comment]
In my case,vue change the row height to 23px,i change back to 24px, it solved.
.handsontable td ,
.handsontable th {  height: 24px;}

**Comment by AMBudnik** [comment]
Example with multi-level data https://jsfiddle.net/126w3amh/ (`7.2.2`, 8.0.0.rc-1 till today works the same)

![bottom rows](https://user-images.githubusercontent.com/10757813/68671137-e6716300-054e-11ea-8dca-3592142d86fa.gif)

**Comment by wojciechczerniak** [comment]
Related #5943

**Comment by AMBudnik** [comment]
When Handsontable doesn't have height defined but a parent has bottom rows are misaligned.

https://jsfiddle.net/23h1jbes/1/ (7.3.0)

![image](https://user-images.githubusercontent.com/10757813/72327714-ea864200-36b1-11ea-96ea-c3330745dfb3.png)

**Comment by wojciechczerniak** [comment]
@AMBudnik Different issue. This task is about not displaying row on the main layer. The second is about positioning it in relation to window (feature that only left and top overlays support ATM)

**Comment by AMBudnik** [comment]
@wojciechczerniak done https://github.com/handsontable/handsontable/issues/6637

**Comment by ashish1697** [comment]
@AMBudnik Is there any update? As our team uses lots of grids with the total at bottom fixed. Still figuring out for the upgrade version to go for.

**Comment by AMBudnik** [comment]
@ashish1697 sorry to say but the fixes for those two cases haven't been scheduled yet. Now we focus on releasing new fixes for beta2 and/or stable 8.0.0 version.

**Comment by aninde** [comment]
I was able to reproduce this problem when Handsontable had a defined size larger than it was able to fill.

https://codesandbox.io/s/double-fixedrows-xj336
![Screenshot 2020-02-04 at 16 04 48](https://user-images.githubusercontent.com/30186818/73756584-1902b480-4768-11ea-99ff-6db6177c0cf1.png)

**Comment by aninde** [comment]
Another case or a different bug?: 
when div with Handsontable is surrounded by other div
https://jsfiddle.net/L1rjuz3e/
![Screenshot 2020-02-10 at 13 05 11](https://user-images.githubusercontent.com/30186818/74148569-07a52680-4c06-11ea-8ed3-24467a00302d.png)

**Comment by AMBudnik** [comment]
Works the same in v 8.0.0http://jsfiddle.net/xkn6Lcgr/

**Comment by wszymanski** [comment]
I did commit https://github.com/handsontable/handsontable/commit/123ee608fce47622cc2861561111f21a4a679548 which fixes the problem for the issue's example by sticking fixed row to the table. 

![image](https://user-images.githubusercontent.com/141330/106132974-ee71ee80-6164-11eb-9a34-208b944bbe99.png)


However, some users may need to stick it to the container.

![image](https://user-images.githubusercontent.com/141330/106133097-17927f00-6165-11eb-9bc4-71b5af31962e.png)

What's more it doesn't touches a problem of setting some CSS values: `height`, `overflow` and Handsontabe's `preventOverflow` settings.

Some requirements should should be rethought yet.

**Comment by aninde** [comment]
I am more than pleased to see that this issue will be fixed by the upcoming version v12.1.0
[original](https://github.com/handsontable/handsontable/issues/4454#issue-248355639) @dosaki demo  http://jsfiddle.net/qoa8ws60/1/ 
[multilevel row's demo](https://github.com/handsontable/handsontable/issues/4454#issuecomment-552870232) https://jsfiddle.net/126w3amh/
[misaligned demo](https://github.com/handsontable/handsontable/issues/4454#issuecomment-574065823) https://jsfiddle.net/vd4567uL/

**Comment by AMBudnik** [comment]
Hi @dosaki I'm happy to notify you that we just released Handsontable v 12.1.0 where this issue is fixed.

As @aninde showed us the bottom rows are no longer in the bottom of the container (they are attached to the table)

![Screenshot 2022-06-28 at 13 58 16](https://user-images.githubusercontent.com/10757813/176172979-02c0d871-25b9-415c-8d12-ff5ca4715bab.png)

CC [ashish1697](https://github.com/ashish1697), [spark889](https://github.com/spark889)
